tool/src/dart/mod.rs

```diff
@@ -919,18 +935,28 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                 let type_name = self.formatter.fmt_type_name(id);
                 format!("{type_name}.values.firstWhere((v) => v._ffi == {var_name})").into()
             }
-            Type::Slice(slice) => if let Some(lt) = slice.lifetime() {
-                let MaybeStatic::NonStatic(lifetime) = lt else {
-                    panic!("'static not supported in Dart");
-                };
-                format!(
-                    "{var_name}._toDart({}Edges)",
-                    lifetime_env.fmt_lifetime(lifetime)
-                )
-            } else {
-                format!("{var_name}._toDart([])")
+            Type::Slice(slice) => {
+                if let Some(lt) = slice.lifetime() {
+                    let MaybeStatic::NonStatic(lifetime) = lt else {
+                        panic!("'static not supported in Dart");
+                    };
+                    format!(
+                        "{var_name}._toDart({}Edges)",
+                        lifetime_env.fmt_lifetime(lifetime)
+                    )
+                    .into()
+                } else {
+                    format!("{var_name}._toDart([])").into()
+                }
+            }
```

Comment:
why did you change this?

Reply:
type inference was failing on the lack of `into()`, apparently it was only working when the last branch is the one with the String.

Reply:
That's weird, both branches are strongly typed, it shouldn't make a difference whether the `into` is on the branches or outside

Reply:
Rust type inference gets really weird around generics like Into. It has some smart heuristics but they often fail like this.

`.into()` is not "strongly typed" there's a lot of inference there, and IIRC there's Into-specific magic that makes Into work okayishly but it's brittle

Reply:
I mean `String` is strongly typed. Before, the if-else had type `String`, which then got `into`d. Now each branch `into`s the `String` separately and that's somehow easier for the compiler!?